boards/stm32l432kc/src/bin/levitation.rs

```diff
@@ -0,0 +1,81 @@
+#![no_std]
+#![no_main]
+
+use defmt::info;
+use advanced_pid::{prelude::*, Pid, PidGain};
+use embassy_executor::Spawner;
+use embassy_stm32::gpio::OutputType;
+use embassy_stm32::timer::low_level::CountingMode;
+use embassy_stm32::timer::simple_pwm::{PwmPin, SimplePwm};
+use embassy_stm32::timer::Channel;
+use embassy_stm32::time::hz;
+use embassy_time::Instant;
+
+use {defmt_rtt as _, panic_probe as _};
+
+const MAX_VOLTAGE: f32 = 500.0; // TBD
+const MAX_CURRENT: f32 = 500.0; // TBD
+const TARGET_HEIGHT: f32 = 10.0; // to be determined by levitation
+
+const GAIN_HEIGHT: PidGain = PidGain{ 
+    // to be determined by levitation
+    kp: 1.0,
+    ki: 0.05,
+    kd: 0.005,
+};
+
+const GAIN_CURRENT: PidGain = PidGain{ 
+    // to be determined by levitation
+    kp: 1.1,
+    ki: 0.12,
+    kd: 0.05,
+};
+
+
+#[embassy_executor::main] 
+async fn main(_spawner: Spawner) {
+
+    let mut pid_height = Pid::new(GAIN_HEIGHT.into());
+    let mut pid_current = Pid::new(GAIN_CURRENT.into());
+
+    let p = embassy_stm32::init(Default::default());
+
+    let green_light = PwmPin::new_ch2(p.PB3, OutputType::PushPull);
+
+    let mut pwm = SimplePwm::new(p.TIM2, None, Some(green_light), None, None, hz(2000), CountingMode::EdgeAlignedUp);
+    pwm.enable(Channel::Ch2);
+
+    let max_duty = pwm.get_max_duty() as f32;
```

Comment:
Factor out the pwm stuff into a `HypedPwm` in `io/pwm.rs` and use that here, so everyone can use PWM as needed